local-cli/init/init.js

```diff
@@ -8,48 +8,104 @@
  */
 'use strict';
 
+const copyProjectTemplateAndReplace = require('../generator/copyProjectTemplateAndReplace');
+const execSync = require('child_process').execSync;
+const fs = require('fs');
+const minimist = require('minimist');
 const path = require('path');
-const TerminalAdapter = require('yeoman-environment/lib/adapter.js');
-const yeoman = require('yeoman-environment');
-
-class CreateSuppressingTerminalAdapter extends TerminalAdapter {
-  constructor() {
-    super();
-    // suppress 'create' output generated by yeoman
-    this.log.create = function() {};
-  }
-}
+const printRunInstructions = require('../generator/printRunInstructions');
+const process = require('process');
+const yarn = require('../util/yarn');
 
 /**
  * Creates the template for a React Native project given the provided
  * parameters:
- *   - projectDir: templates will be copied here.
- *   - argsOrName: project name or full list of custom arguments to pass to the
- *                 generator.
+ * @param projectDir Templates will be copied here.
+ * @param argsOrName Project name or full list of custom arguments
+ *                   for the generator.
  */
 function init(projectDir, argsOrName) {
   console.log('Setting up new React Native app in ' + projectDir);
-  const env = yeoman.createEnv(
-    undefined,
-    undefined,
-    new CreateSuppressingTerminalAdapter()
-  );
 
-  env.register(
-    require.resolve(path.join(__dirname, '../generator')),
-    'react:app'
+  const args = Array.isArray(argsOrName)
+    ? argsOrName // ['AwesomeApp', '--verbose']
+    : [argsOrName].concat(process.argv.slice(4)); // 'AwesomeApp'
+
+  // args is ['AwesomeApp', '--verbose']
+  if (!args || args.lentgh == 0) {
```

Comment:
I assume there is no verbose here?
I think it is not needed, slows down npm even more